File: PrbmMobile/app/src/main/java/it/bitprepared/prbm/mobile/activity/RemoteInterface.java
package it.bitprepared.prbm.mobile.activity;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface RemoteInterface {

    @FormUrlEncoded
    @POST("upload.php")
    Call<String> uploadPrbm(@Field("filename") String filename,
                            @Field("content") String content);

    @FormUrlEncoded
    @POST("image.php")
    Call<String> uploadImage(@Field("name") String filename,
                                   @Field("image") String content);
}

File: app/src/main/java/it/bitprepared/prbm/mobile/activity/RemoteInterface.kt
package it.bitprepared.prbm.mobile.activity

import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

interface RemoteInterface {
    @FormUrlEncoded
    @POST("upload.php")
    suspend fun uploadPrbm(
        @Field("filename") filename: String, @Field("content") content: String
    ): String

    @FormUrlEncoded
    @POST("image.php")
    suspend fun uploadImage(
        @Field("name") filename: String, @Field("image") content: String
    ): String
}
